robot/framesystem/framesystem.go

```diff
@@ -203,6 +206,25 @@ func (svc *frameSystemService) TransformPose(
 	return pose, nil
 }
 
+// TransformPointCloud applies the same pose offset to each point in a single pointcloud and returns the transformed point cloud.
+// if destination string is empty, defaults to transforming to the world frame.
+// Do not move the robot between the generation of the initial pointcloud and the receipt
+// of the transformed pointcloud because that will make the transformations inaccurate.
+func (svc *frameSystemService) TransformPointCloud(ctx context.Context, srcpc pointcloud.PointCloud, srcName, dstName string,
+) (pointcloud.PointCloud, error) {
+	if dstName == "" {
+		dstName = referenceframe.World
+	}
+	// get transform pose needed to get to destination frame
+	sourceFrameZero := referenceframe.NewPoseInFrame(srcName, spatialmath.NewZeroPose())
```

Comment:
Same here, do we need to check for empty `srcName`?

Reply:
went ahead and made a check for source name here too